spec/integration/type/file_spec.rb

```diff
@@ -94,19 +94,39 @@ def get_aces_for_path_by_sid(path, sid)
   end
 
   describe "when ensure is absent" do
-    it "should remove the file if present" do
-      FileUtils.touch(path)
+    before(:each) do
       catalog.add_resource(described_class.new(:path => path, :ensure => :absent, :backup => :false))
-      report = catalog.apply.report
-      expect(report.resource_statuses["File[#{path}]"]).not_to be_failed
-      expect(Puppet::FileSystem.exist?(path)).to be_falsey
     end
 
-    it "should do nothing if file is not present" do
-      catalog.add_resource(described_class.new(:path => path, :ensure => :absent, :backup => :false))
-      report = catalog.apply.report
-      expect(report.resource_statuses["File[#{path}]"]).not_to be_failed
-      expect(Puppet::FileSystem.exist?(path)).to be_falsey
+    context "file is present" do
+      before(:each) do
+        FileUtils.touch(path)
```

Comment:
These are spec/integration, not spec/unit. I don't think they generally mock the filesystem.